src/world/FieldComputations.cpp

```diff
@@ -179,8 +175,8 @@ std::vector<LineSegment> FieldComputations::mergeBlockades(std::vector<LineSegme
     /* If two blockades intersect (in this case, overlap), we take the beginning of the first obstacle and the end of the second obstacle, and put them back in the front of the
      * obstacles vector. The second element gets erased. If they don't intersect, try the next two obstacles. Repeat this procedure until no overlaps are left. */
     std::sort(blockades.begin(), blockades.end(), [](const LineSegment &a, const LineSegment &b) { return a.start.y < b.start.y; });
-    int iterator = 0;
-    while (iterator < static_cast<int>(blockades.size()) - 1) {
```

Comment:
Be careful with these type of things. It often happens in the code that variables values get forgotten and for these situations a static_cast<int> is used.

Reply:
Well static_cast<int>(size_t) can cause overflow.

Reply:
Well static_cast<int>(size_t) can cause overflows so yeah